### DISCUSSION TITLE
Drawing tools: confusion when some tools have sub-tasks and others don't

### DISCUSSION THREAD

**Discussion by shaunanoordin** [zooniverse/front-end-monorepo#6620] [Feature Requests]
## General Bug

Package: `lib-classifier`
Initially discussed in #1677 

Say you have a drawing Task that has multiple Tools, and some tools HAVE sub-tasks but other tools DON'T have sub-tasks. If the active tool DOESN'T have a sub-task, and you click on a mark with a sub-task, then the sub-task popup won't appear.

<img width="968" alt="Screen Shot 2020-06-18 at 17 22 56" src="https://user-images.githubusercontent.com/13952701/85046588-8b5cba00-b188-11ea-910c-10c4f97d2695.png">

_While "Point + no subtasks" Tool is active, clicking on the red or yellow marks (which do have subtasks) doesn't cause the sub-task popup to pop up._

Similarly, if the active Tool selected has a sub-task, but the clicked mark doesn't, an empty sub-task popup appears.

### Testing

Setup:
- Create a workflow with a single drawing task with 2 tools
  - Drawing tool 1: Point tool with a simple single-question sub-task
  - Drawing tool 2: Point tool with no sub-task

Actions:
- Select Tool 1. Create a drawing Mark. (Let's call this Mark 1)
- Select Tool 2. Create a drawing Mark. (Let's call this Mark 2)
- With Tool 2 still selected, click on Mark 1.
  - Expected: the sub-task popup should appear, with a single-question sub-task
  - Actual: no popup appears.
- Select Tool 1. Then, click on Mark 2.
  - Expected: no sub-task popup should appear, because Mark 2 does not have a sub-task associated with it.
  - Actual: the popup appears, but with no sub-tasks (i.e. an empty panel).

Tested on `http://localhost:8080/?env=staging&project=1898&workflow=3396` on macOS 10 + Chrome 83

### Analysis

It seems that the SubTaskPopup works correctly (i.e. sets its visibility to true when a drawing mark is created or selected) _only_ when the _active drawing tool_ has a sub-task associated with it.

❗ Important note: the SubTaskPopup doesn't care what _kind_ of sub-task the active drawing tool has, only whether or not it has one. This means _there's no confusion of sub-tasks when selecting different tools._

For example, if we set up a workflow with 3 drawing tools...
- Drawing tool 1: Point tool with a simple single-question sub-task
- Drawing tool 2: Point tool with no sub-task
- Drawing tool 3: Rectangle tool with a text sub-task

...then the sub-task popup appears correctly (with the correct popups) when Tool 3 is active and we click on Mark 1. 👌  Here, at least, there are no problems.

### Recommendation

As for the source of the problem... Uh, my best guess is that the SubTaskPopup's visibility is somehow tied to checking the subtask value `activeTool` instead of `activeMark`. Or maybe... the activeMark is somehow affected by the activeTool... uh...

Maybe the solution is to always switch ActiveTool to the tool used to create the selected mark, before performing other actions such as showing the popup?

Sorry, I've got nothing.

### Status

Medium-priority bug. Doesn't affect any of our current projects, and we know of the workaround, but needs to be sorted to avoid nasty surprises in the future.

**Comment by srallen** [comment]
Maybe this can be solved with a bit of design. @beckyrother any ideas? Perhaps an icon or tooltip that shows which of the drawing tools has a subtask from the task area?

**Comment by beckyrother** [comment]
I like Shaun's solution of adding '+ subtask' to the tool description! An icon might end up creating more confusion since this isn't necessarily a common thing to encounter
